src/ariadne_wallhug.c

```diff
@@ -724,18 +829,18 @@ long get_next_position_and_angle_required_to_tunnel_creature_to(struct Thing *cr
                 }
                 navi->field_19[0] = ownflag;
 
-                if (get_starting_angle_and_side_of_hug(creatng, &navi->pos_next, &navi->field_D, navi->field_1, 33, a3))
+                if (get_starting_angle_and_side_of_hug(creatng, &navi->pos_next, &navi->angle, navi->field_1, 33, a3))
                 {
                     block_flags = get_hugging_blocked_flags(creatng, &navi->pos_next, 33, a3);
                     set_hugging_pos_using_blocked_flags(&navi->pos_next, creatng, block_flags, thing_nav_sizexy(creatng)/2);
                     if (block_flags == 4)
                     {
-                        if ((navi->field_D == 0) || (navi->field_D == 0x0400))
+                        if ((navi->angle == 0) || (navi->angle == 0x0400))
```

Comment:
I believe the values from the ThingAngles enum work here for all the angles right?